parachain/pallets/control/src/lib.rs

```diff
@@ -312,22 +316,21 @@ pub mod pallet {
 
 		pub fn convert_location(
 			mut location: MultiLocation,
-		) -> Result<(H256, Option<ParaId>, MultiLocation), DispatchError> {
+		) -> Result<(H256, ParaId, MultiLocation), DispatchError> {
 			// Normalize all locations relative to the relay chain.
 			let relay_location = T::RelayLocation::get();
 			location
 				.reanchor(&relay_location, T::UniversalLocation::get())
 				.map_err(|_| Error::<T>::LocationReanchorFailed)?;
 
-			// Only allow Parachain as origin location
-			let para_id = match location {
-				MultiLocation { parents: 0, interior: X1(Parachain(index)) } =>
-					Some((index).into()),
+			let para_id = match location.interior.first() {
+				Some(Parachain(index)) => Some((*index).into()),
 				_ => None,
-			};
+			}
+			.ok_or(Error::<T>::LocationToParaIdConversionFailed)?;
```

Comment:
It will get ParaId from [first Junction](https://github.com/Snowfork/snowbridge/blob/86c3ffc07f6d64d0078b2bd1a0752fcfbe012d3a/parachain/pallets/control/src/lib.rs#L326) of the location, unit tests still work as expected.